src/shared/chat/emoji-dropdown.js

```diff
@@ -95,6 +95,12 @@ export default class EmojiDropdown extends DropdownBase {
         super.showMenu();
         setTimeout(() => this.querySelector('.emoji-search')?.focus());
     }
+
+    hideMenu () {
+        this.chatview.querySelector('converse-emoji-picker')?.disableArrowNavigation();
+        super.hideMenu();
+        setTimeout(() => this.chatview.querySelector('.chat-textarea')?.focus());
```

Comment:
Why did you move this here? I think it's better to leave this line where it was (including the `setTimeout` if that is somehow necessary to fix a race condition).

This, coupled with the above change would mean that it's no longer necessary to overwrite `hideMenu` in this component.

Reply:
Thank you @jcbrand for the consideration of my PR.

There are several use cases with the emojis drop down concerning its closing. All these use cases do not result in calling same code when closing the drop down.
For example, closing the drop down just after its opening by using ESCAPE key, calls `onKeyDown` method in `emoji-picker.js`. But after having navigated into emoji list, closing it by using ESCAPE key or by clicking outside the emoji drop down doesn't call the `onKeyDown` method.
My idea was to find a method which was performed whatever the closing case. It seems to be the case of `hideMenu` method of `emojis-dropdown.js` (by overriding). I found that it was not so bad to centralize the closing stuffs into this method, and so I overrided it. 

That being said, I can try your suggestions.

Reply:
It seems to be same problematics with `_onGlobalKeyDown` method. It is not called whatever the context into which the emoji drop down is closed.

Reply:
Ok. The reason I made my suggestions, is because I'd like to reduce the usage of imperative code where for example we call `querySelector` to call methods on components and because ideally we should decouple components from one another and have the components be as declarative as possible (i.e. more like React and less like jQuery).

In other words, I think it's an anti-pattern to continuously use things like `this.chatview`, because it tightly couples this component to the component referenced by `this.chatview`. The reason why we still use `this.chatview` in various places is because it was the quickest way to turn all the backbone views into web components. But ideally we should try and get the components as independent of one another as possible and remove the usage of `this.chatview` where ever it's still used.

I think eventually we probably will move the emoji picker component into a separate NPM package (and perhaps a separate Github repo), and in that case we won't be able to rely on `this.chatview`.

Reply:
I understand your aim and I agree with it.
Unfortunately, my knowing of the entire of the project, and more generally of JavaScript framework universe, are not enough to propose better for now.
Don't hesitate to close my PR without merging it.
I tried :-)
Thank you @jcbrand.

Reply:
No worries, I've merged your commit and then did some refactoring to bring things more in line with the goals stated above.

Can you please check whether things still work for you as expected?

Reply:
Things are still working as expected :-)
:+1: